spyderlib/plugins/__init__.py

```diff
@@ -131,6 +141,118 @@ def closeEvent(self, event):
         """
         self.plugin_closed.emit()
 
+    def install_tab_event_filter(self, value):
+        """
+        Install an event filter to capture mouse events in the tabs of a
+        QTabBar holding tabified dockwidgets.
+        """
+        bar = None
+        tabbars = self.main.findChildren(QTabBar)
+        for tabbar in tabbars:
+            for tab in range(tabbar.count()):
+                title = tabbar.tabText(tab)
+                if title == self.title:
+                    bar = tabbar
+                    break
+        if bar is not None:
+            self.bar = bar
+            if getattr(self.bar, 'filter', None) is None:
+                self.bar.filter = TabFilter(self.bar, self.main)
+                self.bar.installEventFilter(self.bar.filter)
+
+
+class TabFilter(QObject):
+    """
+    Filter event attached to each QTabBar that holds 2 or more dockwidgets in
+    charge of handling tab rearangement.
+
+    This filter also holds the methods needed for the detection of a drag and
+    the movement of tabs.
+    """
+    def __init__(self, tabbar, main):
+        QObject.__init__(self)
+        self.tabbar = tabbar
+        self.main = main
+        self.moving = False
+        self.start_tab = None
+        self.end_tab = None
+
+    # Helper methods
+    def _get_widget(self, index):

```

Comment:
Well.. we should change the variable in spyder.py that is called widgetlist (other PR) ;-) ... and yes I agree is better with plugins...